File: api/src/endpoint/repository/get_repository_commit_stat.rs
use serde::{Deserialize, Serialize};
use api_derive::EndpointRequest;

use crate::{endpoint::Endpoint, resource::repository::RepositoryCommitStatResource};

pub struct GetRepositoryCommitStat;

impl Endpoint for GetRepositoryCommitStat {
    const PATH: &'static str = "/repository/{owner}/{repo}/commits/{sha}/stat";
    const METHOD: http::Method = http::Method::GET;

    type Request = GetRepositoryCommitStatRequest;
    type Response = GetRepositoryCommitStatResponse;
}

#[derive(EndpointRequest, Debug, Serialize, Deserialize)]
pub struct GetRepositoryCommitStatRequest {}

pub type GetRepositoryCommitStatResponse = Vec<RepositoryCommitStatResource>;

File: gitdot-api-ts/src/endpoint/repository/get-repository-commit-stat.ts
import { z } from "zod";
import { RepositoryCommitStatResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const GetRepositoryCommitStatRequest = z.object({});
export type GetRepositoryCommitStatRequest = z.infer<
  typeof GetRepositoryCommitStatRequest
>;

export const GetRepositoryCommitStatResponse = z.array(
  RepositoryCommitStatResource,
);
export type GetRepositoryCommitStatResponse = z.infer<
  typeof GetRepositoryCommitStatResponse
>;

export const GetRepositoryCommitStat = {
  path: "/repository/{owner}/{repo}/commits/{sha}/stat",
  method: "GET",
  request: GetRepositoryCommitStatRequest,
  response: GetRepositoryCommitStatResponse,
} as const satisfies Endpoint;
export type GetRepositoryCommitStat = typeof GetRepositoryCommitStat;
